### PR TITLE
Fix: Layout update of Asset Page 

### LINKED ISSUE #9349: Enhancement: Update the button layout in Assets Page
**Is your feature request related to a problem? Please describe.**
Update the button layout in Asset page, place all buttons in a same row while preserving the present mobile view

**Describe the solution you'd like**

Present :

Image

Suggested :

Image

Preserving the mobile view:

Image

### PR DESCRIPTION
**Pull request by AdityaJ2305**
## Proposed Changes

- Fixes #9349 
- Change 1
- Change 2
- More?

@ohcnetwork/care-fe-code-reviewers

## Screenshots
Screenshot 2024-12-23 at 8 50 31 PM

Screenshot 2024-12-23 at 8 50 56 PM

## Merge Checklist
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews

## Summary by CodeRabbit

- **New Features**
	- Introduced a new search interface allowing users to search by multiple criteria.
	- Added a button for scanning asset QR codes and creating new assets.
	- Replaced the advanced filter button with a new filter button for improved user interaction.
	- Enhanced dropdown functionality with new components for export options.
	- Added a new type for dropdown item properties to allow for flexible usage.
	- Introduced a new button variant for unstyled buttons.

- **Improvements**
	- Enhanced layout and responsiveness of buttons and filters.
	- Updated button styles for better visual representation in export functionality.
	- Improved user experience with a more organized interface for asset management.

- **Bug Fixes**
	- Maintained error handling for invalid asset IDs during QR scanning.

### CODE REVIEW

**Comment by AdityaJ2305** [comment]
I don’t think the Cypress tests failed due to the changes in this PR. Could someone confirm?

File: src/components/Assets/AssetsList.tsx [hunk 1]
```diff
@@ -29,6 +27,9 @@ import routes from "@/Utils/request/api";
 import request from "@/Utils/request/request";
 import useTanStackQueryInstead from "@/Utils/request/useQuery";
 
+import SearchByMultipleFields from "../Common/SearchByMultipleFields";
+import { Button } from "../ui/button";
```
Comment (Jacobjeevan) [inline src/components/Assets/AssetsList.tsx]: absolute import, other than that, lgtm.

**Review by Jacobjeevan — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
image

- [ ] keep the spacing between the button consistent
- [ ] change the remaining advance filter and export button to use shadcn as well
- [ ] rename advanced filter to filters

**Comment by AdityaJ2305** [comment]
- [x] keep the spacing between the button consistent
- [x]  change the remaining advance filter and export button to use shadcn as well
- [x]  rename advanced filter to filters

@nihal467

File: src/components/Common/Export.tsx [hunk 2]
```diff
@@ -37,7 +44,47 @@
...
+      id={item.options?.id}
+      className={item.options?.className}
+    >
+      <div>
+        {item.options?.icon}
+        <span className="ml-1">{item.label}</span>
+      </div>
+    </DropdownMenuItem>
+  );
+}
```
[the remark this answers is not in the document]
Comment (AdityaJ2305) [reply src/components/Common/Export.tsx]: Argument of type 'AuthorizedForCB | null' is not assignable to parameter of type 'AuthorizedForCB'.

**Comment by AdityaJ2305** [comment]
Hey Team, the linting test is the only one failing due to this PR, and I need some help with it. 😅

**Comment by AdityaJ2305** [comment]
Screenshot 2024-12-24 at 8 00 49 AM
Screenshot 2024-12-24 at 8 01 03 AM

File: src/CAREUI/interactive/FiltersSlideover.tsx [hunk 3]
```diff
@@ -4,6 +4,8 @@ import { useTranslation } from "react-i18next";
 import CareIcon from "@/CAREUI/icons/CareIcon";
 import SlideOver from "@/CAREUI/interactive/SlideOver";
 
+import { Button } from "@/components/ui/button";
+
```
Comment (Jacobjeevan) [inline src/CAREUI/interactive/FiltersSlideover.tsx:8]: Let's remove the ButtonV2 import below (and make sure to sub out any uses of ButtonV2).

File: src/components/Common/Export.tsx [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
+import { ShadcnMenuDropdownItemProps } from "@/components/Common/Menu";
 
 import useExport from "@/hooks/useExport";
+import { useIsAuthorized } from "@/hooks/useIsAuthorized";
 
+import { Anyone } from "@/Utils/AuthorizeFor";
 import request from "@/Utils/request/request";
 import { Route } from "@/Utils/request/types";
 
+import ButtonV2 from "./ButtonV2";
```
Comment (Jacobjeevan) [inline src/components/Common/Export.tsx]: Absolute import here.

**Review by Jacobjeevan — COMMENTED** [review]
lgtm otherwise

File: src/CAREUI/interactive/FiltersSlideover.tsx (hunk 3 above)
Comment (AdityaJ2305) [reply src/CAREUI/interactive/FiltersSlideover.tsx:8]: This file is using ButtonV2 here 👇
[Export.tsx - Line 109](https://github.com/ohcnetwork/care_fe/blob/3b8f7a442473ce828e157697ef8b01d5aefeaa18/src/components/Common/Export.tsx#L109)

Using the Button from Shadcn breaks the UI on other pages like the resource and shifting pages. Here’s an image showing the issue (button beside the title) 👇

Screenshot 2024-12-24 at 4 30 18 PM

Comment (AdityaJ2305) [reply src/CAREUI/interactive/FiltersSlideover.tsx:8]: @Jacobjeevan  Done

**Comment by AdityaJ2305** [comment]
Hey team, all test cases passed after the commit [removed ButtonV2 fully](https://github.com/ohcnetwork/care_fe/pull/9466/commits/a24b580da20d08d6f1bd39d0542b986d0685b965). Following that, only the develop branch was merged, with empty commits added solely to trigger test re-runs.

**Comment by Jacobjeevan** [comment]
> Hey team, all test cases passed after the commit [removed ButtonV2 fully](https://github.com/ohcnetwork/care_fe/pull/9466/commits/a24b580da20d08d6f1bd39d0542b986d0685b965). Following that, only the develop branch was merged, with empty commits added solely to trigger test re-runs.

Just ping the team on slack to rerun the tests next time 😅

**Comment by AdityaJ2305** [comment]
> > Hey team, all test cases passed after the commit [removed ButtonV2 fully](https://github.com/ohcnetwork/care_fe/pull/9466/commits/a24b580da20d08d6f1bd39d0542b986d0685b965). Following that, only the develop branch was merged, with empty commits added solely to trigger test re-runs.
> 
> Just ping the team on slack to rerun the tests next time 😅
Actually, Javid once asked the same thing in the Slack channel, and Rithvik mentioned pushing empty commits to re-run test cases. That’s why I have a couple of empty commits, just for triggering the tests again.  😊

Edit: Passed 🥳

File: src/components/Assets/AssetsList.tsx [hunk 5]
```diff
@@ -317,9 +319,54 @@ const AssetsList = () => {
...
+              >
+                <CareIcon icon="l-plus-circle" className="text-lg mr-2" />
+                <span>{t("create_asset")}</span>
+              </Button>
+            </div>
+          </div>
+
+          {/* Advanced Filter Button */}
+          <div className="w-full md:w-auto">
+            <AdvancedFilterButton
```
Comment (rithviknishad) [inline src/components/Assets/AssetsList.tsx]: lgtm, minor thing: let's rename the component too.

```suggestion
            <FilterButton
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by AdityaJ2305** [comment]
> ```diff
>   <Button
> +              variant="destructive"
> +              className="flex items-center gap-2"
> ```

Hey Team! I’ll be addressing this topic in the end-of-day meeting. 😊

**Comment by nihal467** [comment]
LGTM

**Comment by rithviknishad** [comment]
no longer relevant in new architecture

**Pull request closed without merging**